automation/helpers.go

```diff
@@ -0,0 +1,92 @@
+package automation
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"time"
+
+	ndbv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+func ReadYAMLFile(filename string) ([]byte, error) {
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read YAML file: %v", err)
+	}
+
+	return yamlFile, nil
+}
+
+func ConvertToSecret(data []byte) (secret *corev1.Secret, err error) {
+	secret = &corev1.Secret{}
+	jsonData, err := yaml.YAMLToJSON(data)
+	if err != nil {
+		log.Println(err)
+	}
+	err = json.Unmarshal(jsonData, secret)
+	if err != nil {
+		log.Println(err)
+	}
+	return
+}
+
+func ConvertToDatabase(data []byte) (database *ndbv1alpha1.Database, err error) {
```

Comment:
all of these utility functions essentially unmarshal a byte stream into an object of interest (secret, pod, database, etc).
can we just have 1 func that does all this? it takes "type" as an argument and maybe we can do some magic with the reflect package. can you pls see if that is possible?